src/FsAutoComplete.Core/FCSPatches.fs

```diff
@@ -235,3 +235,46 @@ type FSharpParseFileResults with
       match scope.ParseTree with
       | Some input -> SynExprAppLocationsImpl.getAllCurriedArgsAtPosition pos input
       | None -> None
+
+  member scope.TryRangeOfNearestOuterBindingContainingPos pos =
+    let identInBinding binding =
+        match binding with
+        | SynBinding.Binding (_, _, _, _, _, _, _, headPat, _, _, _, _) ->
+            match headPat with
+            | SynPat.LongIdent (longIdentWithDots, _, _, _, _, _) ->
+                Some longIdentWithDots.Range
+            | SynPat.Named(_, ident, false, _, _) ->
+                Some ident.idRange
+            | _ ->
+                None
+
+    let rec walkLetOrUse expr workingRange =
+        match expr with
```

Comment:
thinking about this some more I may have more work to do, since the `LetOrUse` could come after some other construct and thus be nested. But for now it's probably okay and covers most cases